plonk-core/src/proof_system/widget/lookup.rs

```diff
@@ -210,71 +210,78 @@ where
     }
 }
 
-/// Permutation Verifier Key
+/// LookUp Verifier Key
 #[derive(CanonicalDeserialize, CanonicalSerialize, derivative::Derivative)]
 #[derivative(
-    Clone(bound = ""),
-    Debug(bound = "PCC: std::fmt::Debug"),
-    Eq(bound = "PCC: Eq"),
-    PartialEq(bound = "PCC: PartialEq")
+    Clone,
+    Copy(bound = "PC::Commitment: Copy"),
+    Debug(bound = "PC::Commitment: std::fmt::Debug"),
+    Eq(bound = "PC::Commitment: Eq"),
+    PartialEq(bound = "PC::Commitment: PartialEq")
 )]
-pub struct VerifierKey<PCC>
+pub struct VerifierKey<F, PC>
 where
-    PCC: PCCommitment + Default,
+    F: PrimeField,
+    PC: PolynomialCommitment<F, DensePolynomial<F>>,
 {
     /// Lookup Selector Commitment
-    pub q_lookup: PCC,
+    pub q_lookup: PC::Commitment,
 }
 
-impl<PCC> VerifierKey<PCC>
+impl<F, PC> VerifierKey<F, PC>
 where
-    PCC: PCCommitment + Default,
+    F: PrimeField,
+    PC: PolynomialCommitment<F, DensePolynomial<F>>,
 {
     /// Computes the linearisation commitments.
-    pub fn compute_linearisation_commitment<F: FftField>(
+    pub fn compute_linearisation_commitment(
         &self,
         scalars: &mut Vec<F>,
-        points: &mut Vec<PCC>,
+        points: &mut Vec<PC::Commitment>,
         evaluations: &ProofEvaluations<F>,
         (delta, epsilon, zeta): (F, F, F),
-        lookup_sep: F,
         l1_eval: F,
-        z2_comm: PCC,
-        h1_comm: PCC,
+        z2_comm: PC::Commitment,
+        h1_comm: PC::Commitment,
     ) {
-        let f = {
-            let f_0 = evaluations.wire_evals.a_eval
-                - evaluations.lookup_evals.f_eval
-                + zeta * evaluations.wire_evals.b_eval
-                + zeta * zeta * evaluations.wire_evals.c_eval
-                + zeta * zeta * zeta * evaluations.wire_evals.d_eval;
-            f_0 * lookup_sep
-        };
+        // f =q_lookup * (lc([a_eval, b_eval, c_eval, d_eval] , zeta) - f_eval)
+        // lookup_sep is the eval of q_lookup and should be in the proof
+
```

Comment:
I don't think that is true. q_lookup doesn't work as a selector on the lookup permutation terms which  need a separator